core/src/main/scala/org/json4s/Extraction.scala

```diff
@@ -137,7 +138,7 @@ object Extraction {
             val vvv = if (fieldSerializer.includeLazyVal) loadLazyValValue(a, nn, vv) else vv
             addField(nn, vvv, obj)
           }
-        } else if (ctorParams contains prop.name) addField(n, fieldVal, obj)
+        } else if ((ctorParams contains prop.name) && (methods contains prop.name.replaceAll("-", "\\$minus"))) addField(n, fieldVal, obj)
```

Comment:
I think this is not enough. You should handle all symbol names. (Maybe use [NameTransformer](https://github.com/scala/scala/blob/v2.12.1/src/library/scala/reflect/NameTransformer.scala) ?)

I added more tests https://github.com/json4s/json4s/commit/0380e196d5b4c2e3590c269ba68754cd9ca2b9ce